tests/specs/lemmas.k

```diff
@@ -88,118 +94,105 @@ module LEMMAS
        andBool 0 <=Int V2 andBool V2 <Int pow256
       [concrete]
 
-    rule keccakIntList(V:Int .IntList) => hash1(V)
-    rule keccakIntList(V1:Int V2:Int .IntList) => hash2(V1, V2)
+    rule keccakIntList(V:Int .IntList) => hash1(V)               [simplification]
+    rule keccakIntList(V1:Int V2:Int .IntList) => hash2(V1, V2)  [simplification]
 
     // for terms came from bytecode not via #hashedLocation
     rule keccak(WS) => keccakIntList(byteStack2IntList(WS))
-      requires ( notBool #isConcrete(WS) )
-       andBool ( #sizeByteArray(WS) ==Int 32 orBool #sizeByteArray(WS) ==Int 64 )
+      requires ( #sizeByteArray(WS) ==Int 32 orBool #sizeByteArray(WS) ==Int 64 )
+      [simplification]
 
-    rule N +Int 0 => N
+    rule N +Int 0 => N  [simplification]
 
-    rule N -Int 0 => N
+    rule N -Int 0 => N  [simplification]
 
-    rule 1 *Int N => N
-    rule N *Int 1 => N
-    rule 0 *Int _ => 0
-    rule _ *Int 0 => 0
+    rule 1 *Int N => N  [simplification]
+    rule N *Int 1 => N  [simplification]
+    rule 0 *Int _ => 0  [simplification]
+    rule _ *Int 0 => 0  [simplification]
 
-    rule N /Int 1 => N
+    rule N /Int 1 => N  [simplification]
 
-    rule 0 |Int N => N
-    rule N |Int 0 => N
-    rule N |Int N => N
+    rule 0 |Int N => N  [simplification]
+    rule N |Int 0 => N  [simplification]
+    rule N |Int N => N  [simplification]
 
-    rule 0 &Int N => 0
-    rule N &Int 0 => 0
-    rule N &Int N => N
-
-    //orienting symbolic term to be first, converting -Int to +Int for concrete values.
-    rule I +Int B => B          +Int I when #isConcrete(I) andBool notBool #isConcrete(B)
-    rule A -Int I => A +Int (0 -Int I) when notBool #isConcrete(A) andBool #isConcrete(I)
-
-    rule (A +Int I2) +Int I3 => A +Int (I2 +Int I3) when notBool #isConcrete(A) andBool #isConcrete(I2) andBool #isConcrete(I3)
-
-    rule I1 +Int (B +Int I3) => B +Int (I1 +Int I3) when #isConcrete(I1) andBool notBool #isConcrete(B) andBool #isConcrete(I3)
-    rule I1 -Int (B +Int I3) => (I1 -Int I3) -Int B when #isConcrete(I1) andBool notBool #isConcrete(B) andBool #isConcrete(I3)
-    rule (I1 -Int B) +Int I3 => (I1 +Int I3) -Int B when #isConcrete(I1) andBool notBool #isConcrete(B) andBool #isConcrete(I3)
-
-    rule I1 +Int (I2 +Int C) => (I1 +Int I2) +Int C when #isConcrete(I1) andBool #isConcrete(I2) andBool notBool #isConcrete(C)
-    rule I1 +Int (I2 -Int C) => (I1 +Int I2) -Int C when #isConcrete(I1) andBool #isConcrete(I2) andBool notBool #isConcrete(C)
-    rule I1 -Int (I2 +Int C) => (I1 -Int I2) -Int C when #isConcrete(I1) andBool #isConcrete(I2) andBool notBool #isConcrete(C)
-    rule I1 -Int (I2 -Int C) => (I1 -Int I2) +Int C when #isConcrete(I1) andBool #isConcrete(I2) andBool notBool #isConcrete(C)
-
-    rule I1 &Int (I2 &Int C) => (I1 &Int I2) &Int C when #isConcrete(I1) andBool #isConcrete(I2) andBool notBool #isConcrete(C)
```

Comment:
you just dropped this set of lemmas, right?  i think it is fine for now, and revisit later if you need.

Reply:
Yes. `#isConcrete` will probably never work on the Haskell backend.